### DISCUSSION TITLE
Trouble changing PDF font using `fontfamily`

### DISCUSSION THREAD

**Discussion by kjohnsen** [quarto-dev/quarto-cli#2343] [Q&A] [answered]
Hello, I'm using the default PDF engine and am trying to change the font using the `fontfamily` option, which [supposedly takes a font package](https://pandoc.org/MANUAL.html#variables-for-latex:~:text=font%20package%20for%20use%20with%20pdflatex) name as an argument. This works with `fontfamily: libertinus` but not with `fontfamily: mathpazo`. I can't figure out why. I've tried to mess with `mainfont` as well, but can't get the Palatino font I'm after.

**Comment by cderv** [comment]
Is there any error message that is thrown and give hints ? 

From https://www.ctan.org/pkg/mathpazo, 
> The Pazo Math fonts are a family of PostScript fonts suitable for typesetting mathematics in combination with the Palatino family of text fonts. 

So isn't it only for math rendering  ? This one is also for Palatino https://www.ctan.org/pkg/domitian

Also not that `fontfamily` is only for pdflatex engine I think - https://quarto.org/docs/reference/formats/pdf.html#fonts
About `fontfamily`
> Font package to use when compiling a PDf with the pdflatex pdf-engine.

Default engine is **xelatex** with Quarto, and you need to [change it](https://quarto.org/docs/output-formats/pdf-engine.html#alternate-pdf-engines) or use the correct font option (`mainfont` I think with xelatex). 

Hope it helps.

**Reply by kjohnsen** [reply]
So I did notice that `fontfamily` was only for pdflatex, but then why did setting it to `libertinus` work on xelatex? 

Looking at [this](https://tex.stackexchange.com/a/198949/110385) StackExchange answer, I was able to use a hacky workaround, setting fonts in the LaTeX header rather than through pandoc's options. I have no understanding of why this works, but it did the trick without having to switch PDF engines:
```tex
\usepackage{fontspec}
\usepackage[T1]{fontenc}
\usepackage{mathpazo}
\setsansfont{Open Sans}
```

`\usepackage{mathpazo}` actually sets the serif font to Palatino; it doesn't just add the math font.

**Reply by kjohnsen** [reply]
My rationale for doing this was to get both the Palatino and Pazo Math fonts, which work well together. I guess the less hacky way of doing this with xelatex would be to make sure I have the fonts installed on my system and use `mainfont` and `mathfont` options

**Reply by cderv** [reply]
> \usepackage{mathpazo} actually sets the serif font to Palatino; it doesn't just add the math font.

good to know ! thanks !

**Comment by kjohnsen** [comment, marked as answer]
Update: yep, this worked fine with xelatex after installing these fonts on my system and refreshing the font cache.
```yml
format:
  pdf:
    mainfont: TeX Gyre Pagella
    mathfont: TeX Gyre Pagella Math
    sansfont: Open Sans
```

**Reply by cderv** [reply]
I think that is the correct way to do that with Pandoc, at least how the Pandoc's template expects it. Unless you switch fro xelatex to pdflatex

I think you can mark this as the answer

**Reply by kjohnsen** [reply]
Thanks!
